src/ESP32PWM.cpp

```diff
@@ -8,6 +8,14 @@
 #include <ESP32PWM.h>
 #include "esp32-hal-ledc.h"
 
+#if !defined(ESP_ARDUINO_VERSION)
```

Comment:
this will mess up toolchain detection on other libraries imported after this header. Just add the version detection gaurdes around each section of code where differing behavior is needed for different versions of the toolchian

Reply:
Sorry I do not really understand.
I think, this code is not in a header, so it is strictly locally to to code below.